src/components/connectAdvanced.js

```diff
@@ -2,6 +2,7 @@ import hoistStatics from 'hoist-non-react-statics'
 import invariant from 'invariant'
 import { Component, createElement } from 'react'
 import { polyfill } from 'react-lifecycles-compat'
+import * as ReactIs from 'react-is'
```

Comment:
For the purposes of tree shaking, you should import only the `isValidElementType` function.

Reply:
@timdorr I did try, it fails with: 

```bash
src/index.js → dist/react-redux.js...
[!] Error: 'isValidElementType' is not exported by node_modules/react-is/index.js
https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module
src/components/connectAdvanced.js (5:9)
3: import { Component, createElement } from 'react'
4: import { polyfill } from 'react-lifecycles-compat'
5: import { isValidElementType } from 'react-is'
            ^
```

Reply:
any clues? here is the package: https://github.com/facebook/react/tree/master/packages/react-is

Reply:
For reference, here's the actual built file from the NPM package: https://unpkg.com/react-is@16.4.1/cjs/react-is.development.js .

The entire built `react-is` package is CJS, so I'm not sure there's any tree-shaking that's going to happen.

Reply:
AFAIK, Rollup and Webpack are working on tree-shaking/DCE for CJS bundles (well, all code, really).

But `react-is` is built exactly like `react`, so I'm not sure why that doesn't work. Maybe our build tools need an upgrade?

Reply:
@timdorr to be more specific, it happens when `npm run build:umd` is executed (consequently `build:umd:min` also fails).

Reply:
@timdorr @markerikson I updated every dependency and the issue is still there. I don't think we are going to be able to do that. You are welcome to try.

<img width="730" alt="screen shot 2018-07-12 at 11 41 42 am" src="https://user-images.githubusercontent.com/8354571/42653005-f1362d68-85c8-11e8-96b5-d3e1d86506d5.png">

Reply:
Just a comment, by doing that, the tests fail. I only did it because I tried updating a single dependency at the time and it didn't work.

Reply:
@timdorr I don't want this PR to get stale, is there anything you recommend me doing? E.g. maybe creating a PR (or issue) to react-is so that it exports individual modules to enable tree-shaking

Reply:
as mentioned in https://github.com/reduxjs/react-redux/issues/914#issuecomment-377406113 it's not yet possible to import specific portions of react-is, but if you fix the build of react-is, it would then become possible

Reply:
@timdorr @cellog ok, here is my attempt https://github.com/facebook/react/pull/13250

Reply:
Cool. Let's see if they can get that out soon enough. I'm not particularly worried about this aging quickly, as this part of the code isn't often touched.

Reply:
I closed it because it was not the expected solution.